### ISSUE TITLE
Connected but not receiving on subscibed topic

### ISSUE THREAD

**Issue by tomy983** [grafana/mqtt-datasource#60]
I got the plugin installed (using grafana 9.3.1) on ubuntu 20.04
Added the data source, (using password auth) and got it correctly connected.
![image](https://user-images.githubusercontent.com/11900154/207656020-9564b115-34c7-446c-90de-5e0b80a45473.png)

No data is received on test topic (to which I publish an integer as raw data).
![image](https://user-images.githubusercontent.com/11900154/207655688-6ee75e61-96d4-4b99-a45b-2e409a78b62d.png)

Tried also with `/test` an also using json with no difference. Data is no coming. Looked for info in grafana logs but could not find anything relevant (do I need to set up logging somewhere?)

**Comment by edizaziz-dev** [comment]
Just tagging along to see if any one else is experiencing this also, I certainly am. 

My experience so far is, I know the mqtt broker is working because I can see it in node-red. 

In Grafana, in the explore area, I checked the browser's network tab and I can see successful posts to "query" with a valid payload to the server but the table resolves to 0 results with no further responses from the server or the table just keep loading forever. 

I'm running it all under home assistant which in a container on proxmox.

**Comment by seeul8er** [comment]
I had the same issue. I changed from TCP to WebSocket and now it works. I can see the data from the subscribed topics.
Seems like something is broken with TCP?

![image](https://user-images.githubusercontent.com/24637325/208435707-957dc998-8a3f-4787-93f0-2d22e8fa9bd4.png)

**Comment by tomy983** [comment]
I don't have websockets enabled, so obviously this is not working for me. 
Are there any debug log I could activate to look further into why this isn't working?

**Comment by tomy983** [comment]
Right, finally got around to test also WebSocket. Unfortunately the result is exactly the same as [the first post](https://github.com/grafana/mqtt-datasource/issues/60#issue-1497024683)..

**Comment by OlivierGre** [comment]
I have the same problem.

**Comment by stian-01** [comment]
I have the same problem... I have Grafana 9.3.6, using Aedes MQTT broker in Node-RED (which I know works as I use it for several things).
In the Grafana log it says "Connecting", but never "Connected" if it's supposed to say that. It also says "Disconnected" and then "Connecting" if I redeploy in Node-RED or something (i.e. it should work). And the Aedes broker also states that the Grafana client is connected correctly, like any other connected client.
I still get nothing, regardless of the topics I subscribe to or the messages being sent

**Comment by jforge** [comment]
Same issue on Grafana 9.3.1 and different brokers: Mosquitto, HiveMQ, VerneMQ.
All brokers are 100% working as expected for unencrypted tcp/1883, tls/8883 and wss over 443 with a /mqtt context (VerneMQ).

Also:
- can't use tls or secure websockets without perfectly deployed certs, because there is no option to accept all servers certs 
- can't use mutual tls, because of missing configuration settings

For now, I need to use the regular telegraf/mqtt to influxdb to grafana approach as usual, but I would actually like to use this MQTT plugin directly without TIG/TICK approaches.

**Comment by avfuturist** [comment]
Any progress here? I'm having the same issue...just can't get the plugin to receive any data. Using Grafana in a container and the EMQ public broker over TCP and/or Websocket.

**Comment by cicciopizza** [comment]
I have the same problem

**Comment by ryanshow** [comment]
I'm also experiencing this exact same issue. No entries in any log that I can find suggesting a failure is happening, the query simply never returns anything.

**Comment by mcules** [comment]
Same here. It's connected to the broker, but no messages arrive.

**Comment by chrispi68** [comment]
Same problem here. I think this project is totally broken.

**Comment by daan0308** [comment]
Same problem here. We can get it working locally but not over the network. It is weird to see why this is, since the setup is completely the same.

**Comment by jforge** [comment]
> Same problem here. We can get it working locally but not over the network. It is weird to see why this is, since the setup is completely the same.

@daan0308 could you please describe, how you get this working locally? This is interesting and one step forward.

**Comment by daan0308** [comment]
My issue was fortunately not related to the above issue, but related to proxy settings and the way grafana works. 

Grafana live (which is the implementation this adapter uses to stream its data) uses websockets to connect to the Grafana web interface. My instance was setup behind a proxy, which did not have http to websocket upgrades configured correctly. 

The weird thing is, and this should be more of a Grafana issue, is that Grafana should show an upgrade/http error instead of 'no data'. That would make it way more easy to debug.

**Comment by Jaroslav-Dvorak** [comment]
Try this in Grafana addon configuration - add env_vars.
![Clipboard01](https://github.com/grafana/mqtt-datasource/assets/56601456/95273754-09a9-4e38-b5ac-d7b5a5bdb5c1)
